model/test_results_test.go

```diff
@@ -80,6 +80,79 @@ func TestTestResultsFind(t *testing.T) {
 	})
 }
 
+func TestTestResultsFindByTaskID(t *testing.T) {
+	env := cedar.GetEnvironment()
+	db := env.GetDB()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	defer func() {
+		assert.NoError(t, db.Collection(testResultsCollection).Drop(ctx))
+	}()
+
+	tr1 := getTestResults()
+	tr1.Info.Execution = 0
+	_, err := db.Collection(testResultsCollection).InsertOne(ctx, tr1)
+	require.NoError(t, err)
+
+	tr2 := getTestResults()
+	tr2.Info.TaskID = tr1.Info.TaskID
+	tr2.Info.Execution = 1
+	_, err = db.Collection(testResultsCollection).InsertOne(ctx, tr2)
+	require.NoError(t, err)
+
+	t.Run("DNE", func(t *testing.T) {
```

Comment:
Good tests! On tests that expect an error (not found, etc.) check that population is still false